File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/stream/StreamViewModelFactory.java
package org.dvbviewer.controller.data.stream;

import android.app.Application;
import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;

public class StreamViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private Application mApplication;
    private StreamRepository mRepo;

    public StreamViewModelFactory(Application application, StreamRepository repo) {
        mApplication = application;
        mRepo = repo;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new StreamViewModel(mApplication, mRepo);
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/stream/StreamViewModelFactory.kt
package org.dvbviewer.controller.data.stream

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class StreamViewModelFactory(private val mApplication: Application, private val mRepo: StreamRepository) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return StreamViewModel(mApplication, mRepo) as T
    }

}